src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs

```diff
@@ -44,7 +44,23 @@ internal static string EmitQueryParameterPreparation(this EndpointParameter endp
                         {
                             wasParamCheckFailure = true;
                         }
-                        var {{endpointParameter.HandlerArgument}} = {{endpointParameter.Name}}_raw.ToString();
+                        var {{endpointParameter.Name}}_temp = {{endpointParameter.Name}}_raw.ToString();
+""");
+        }
+
+        if (endpointParameter.IsParsable)
+        {
+            var parsingBlock = endpointParameter.ParsingBlockEmitter($"{endpointParameter.Name}_temp", $"{endpointParameter.Name}_parsed_temp");
```

Comment:
Have we discussed about using the [System.CodeDom.Compiler.IndentedTextWriter](https://source.dot.net/System.Private.CoreLib/R/bae755007e6f4473.html)?